// @codux-boards/react-bootstrap/boards/Bootstrap/Carousel/carousel.board.tsx
import { createBoard } from '@wixc3/react-board';
import Carousel from 'react-bootstrap/Carousel';

// board cover
import cover from './carousel.svg';

export default createBoard({
    name: 'Carousel',
    Board: () => {
        return (
            <Carousel interval={null} style={{ width: '500px' }}>
                <Carousel.Item>
                    <img
                        src="https://static.wixstatic.com/media/68d3a9_e54bc05fc07647e189baeee0441eb104~mv2.jpg"
                        alt="First slide"
                        style={{
                            objectFit: 'cover',
                            width: '100%',
                            height: '500px',
                        }}
                    />
                    <Carousel.Caption>
                        <h3>First slide label</h3>
                        <p>
                            Nulla vitae elit libero, a pharetra augue mollis
                            interdum.
                        </p>
                    </Carousel.Caption>
                </Carousel.Item>
                <Carousel.Item>
                    <img
                        src="https://static.wixstatic.com/media/68d3a9_e54bc05fc07647e189baeee0441eb104~mv2.jpg"
                        alt="Second slide"
                        style={{
                            objectFit: 'cover',
                            width: '100%',
                            height: '500px',
                        }}
                    />

                    <Carousel.Caption>
                        <h3>Second slide label</h3>
                        <p>
                            Lorem ipsum dolor sit amet, consectetur adipiscing
                            elit.
                        </p>
                    </Carousel.Caption>
                </Carousel.Item>
                <Carousel.Item>
                    <img
                        src="https://static.wixstatic.com/media/68d3a9_e54bc05fc07647e189baeee0441eb104~mv2.jpg"
                        alt="Third slide"
                        style={{
                            objectFit: 'cover',
                            width: '100%',
                            height: '500px',
                        }}
                    />

                    <Carousel.Caption>
                        <h3>Third slide label</h3>
                        <p>
                            Praesent commodo cursus magna, vel scelerisque nisl
                            consectetur.
                        </p>
                    </Carousel.Caption>
                </Carousel.Item>
            </Carousel>
        );
    },
    tags: ['carousel', 'slide'],
    cover,
});
